<a href="https://colab.research.google.com/github/EEdwardsA/DS-Unit-1-Sprint-2-Statistics/blob/master/module3/LS_DS_123_Confidence_Intervals_Assignment_Elizabeth_new.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Confidence Intervals

Soft drinks like Coke and Pepsi are manufactured to have a standard caffeine content. For example, a 12-oz serving of Coke has 34mg of caffeine, and a 12-oz serving of Pepsi has 37.6mg of caffeine. However, fountain soft drinks are typically mixed in individual restaurant dispensers, so it is more difficult to maintain a standard level of caffeine per serving. In this study, researchers randomly sampled Coke, Diet Coke, Pepsi, and Diet Pepsi at a set of franchise restaurants and measured the caffeine content in 12oz of each soft drink2. The data is found in the Soda.xlsx dataset.

Because individuals can be sensitive to caffeine – and because the manufacturers are interested in product consistency – we wish to estimate the mean caffeine content in 12oz of Coke served in franchise restaurants using a 95% confidence interval. 

You can find the Coke data here: 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'

The first variable is the sample ID and the second variable is the caffeine content in the 12-oz sample measured in mg.

Source: A.N. Garand and L.N. Bell (1997). "Caffeine Content of Fountain and Private-Label Store Brand Carbonated Beverages," Journal of the American Dietetic Association, Vol. 97, #2, pp. 179-182.




###1) Load the dataset and print the first few rows

In [1]:
#Import statements

import pandas as pd
import numpy as np

coke = pd.read_csv('https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv')

In [2]:
#Print the top 5 rows

coke.head()

Drink  Caffeine
0      1     47.32
1      2     43.78
2      3     48.12
3      4     43.25
4      5     46.42

###2) Calculate the mean, SD, SE caffeiene content and n for the sample.  Summarize your results in a sentence or two.

In [12]:
#mean
mean = coke['Caffeine'].mean()
mean

37.9402

In [13]:
#standard deviation
sd = coke['Caffeine'].std()
sd

5.243756828216712

In [14]:
#standard error
se = coke['Caffeine'].sem()
se

0.7415792024250598

In [15]:
#sample size (n)
n = coke['Caffeine'].count()
n

50

In a sample size of 50, the average (mean) caffeine content is 37.94 milligrams. The standard error is 0.74 and the standard deviation is 5.24 mg.

###3) Find t* for a 95% confidence interval.  

Use the starter code below and fill in the degrees of freedom.

In [8]:
from scipy.stats import t

#Don't worry too much about where the 0.975 comes from.  It has to do
#with wanting to determine the *middle* 95% of the t-distribution
#We're going to learn
#how to calculate a 95% CI this easy way in just a minute.

#Recall that n = 223 for the body temp problem.
t_star = t.ppf(0.975,df=49)
print('t_star =', t_star)

t_star = 2.009575234489209


###4) Calculate the margin of error for a 95% confidence interval for the mean caffeine content in a 12-oz Coke.



In [16]:
#Calculations here

# Margine of error is t_star * (sd/(n**(1/2)))

moe = t_star * (sd/(n**(1/2)))

moe

1.49025919960566

The margin of error is 1.49.


###5) Calculate a 95% CI for the mean caffeine content in a 12-oz fountain Coke with the CI formula using the summary statistics and t* that you calculated above.

In [19]:
37.94 + 1.49

39.43

In [20]:
37.94 - 1.49

36.449999999999996

###6) Calculate a 95% CI for the mean caffeiene content in a 12-oz fountain Coke using the t-interval function in Python.

In [21]:
#Calculations here

t.interval(alpha=0.95, df=49, loc=37.9402, scale=0.7415792024250598)

(36.44994080039434, 39.43045919960566)

###7) Compare the two confidence intervals you calculated.  Do they match?  Should they?

They match, but not perfectly, due to rounding.

###8) Interpret the meaning of the 95% confidence interval for the mean caffeiene content in a 12-oz fountain Coke. in a sentence or two.

We are 95% confident that a 12 oz. Coke dispensed from a soda fountain at a franchise restaurant has between 36.45 and 39.43 mg of caffeine.

###9) Using the t-interval Python function, calculate a 90% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [22]:
t.interval(alpha=0.90, df=49, loc=37.9402, scale=0.7415792024250598)

(36.696904726749196, 39.1834952732508)

We are 90% confident that a 12 oz. Coke dispensed from a soda fountain at a franchise restaurant contained between 36.70 and 39.18 mg of caffeine.

This is more precise but less accurate than our 95% confidence interval.

###10) Using the t-interval Python function, calculate a 99% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [23]:
t.interval(alpha=0.99, df=49, loc=37.9402, scale=0.7415792024250598)

(35.95280335285685, 39.92759664714315)

We are 99% confident that a 12 oz. Coke dispensed from a soda fountain at a franchise restaurant contains between 35.95 and 39.93 mg of caffeine.

This is more accurate but less precise than our 95% confidence interval.

## Stretch goals:

###1) The correspondence between confidence intervals and hypothesis tests.

Read [this](https://https://blog.minitab.com/blog/adventures-in-statistics-2/understanding-hypothesis-tests-confidence-intervals-and-confidence-levels#:~:text=If%20a%20hypothesis%20test%20produces,corresponding%20confidence%20level%20is%2095%25.&text=If%20the%20confidence%20interval%20does,the%20results%20are%20statistically%20significant.) article about the correspondence between confidence intervals and hypothesis tests.  Feel free to read the whole article, but the relevant part can be found under the heading Why P Values and Confidence Intervals Always Agree About Statistical Significance.

Imagine you work for quality control at Coke and are tasked with making sure that the caffeiene content in the fountain beverages served in restaurants is the same as in a 12-oz can of Coke (34mg).  If you believe that the mean caffeiene content in fountain coke is not 34mg, you must re-train the franchise managers to make sure the Coke served has the correct caffeiene level.

Based on the confidence interval you calculated in the assignment, do you believe that the mean caffeiene content is statistically significantly different from 34 mg in a 12-oz serving?


Yes, the mean caffeine content is statistically significantly different from 34 mg. Even at the 99% confidence level, the 34 mg value, which is our null hypothesis value, is not contained within the interval of 35.95 to 39.93 mg. We will have to re-train the franchise managers to get them to stop overcaffeinating customers who purchase 12 oz. servings of Coke.

###2) If we increased the sample size from 50 to 100 but the sample mean and SD remained the same, describe **two** ways the margin of error would change.  Would the margin of error become smaller or larger?

With a larger sample size, our margin of error would decrease.

In [24]:
#Calculations here.
t_star = t.ppf(0.975,df=99)
print('t_star =', t_star)



t_star = 1.9842169515086827


In [25]:
# Margine of error is t_star * (sd/(n**(1/2)))

moe = t_star * (sd/(100**(1/2)))
moe 

1.0404751188137005

Compare MEs here.
The former margin of error, with a sample size of 50, was 1.49. The new margin of error, with a sample size of 100, is smaller, at 1.04.